### PR TITLE
link to boost::headers

### PR DESCRIPTION
**Pull request by mgovers**
Explicitly link to Boost::headers to prevent confusion about whether we use the header-only library of Boost or not.

Bumps CMake minimum required version to `3.15`

### CODE REVIEW

File: cmake/export_power_grid_model.cmake [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #
 # SPDX-License-Identifier: MPL-2.0
 
-cmake_minimum_required (VERSION 3.9)
+cmake_minimum_required (VERSION 3.15)
```
Comment (TonyXiang8787) [inline cmake/export_power_grid_model.cmake:5]: Do you also need 3.15 for export target? I think for the users of C-API, we don't need this cmake version right?

File: cmake/pgm_version.cmake [hunk 2]
```diff
@@ -2,7 +2,7 @@
 #
 # SPDX-License-Identifier: MPL-2.0
 
-cmake_minimum_required (VERSION 3.9)
+cmake_minimum_required (VERSION 3.15)
```
Comment (TonyXiang8787) [inline cmake/pgm_version.cmake:5]: Same here

File: tests/package_tests/CMakeLists.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model project <dynamic.grid.calculation@alliander.com>
 #
 # SPDX-License-Identifier: MPL-2.0
-cmake_minimum_required (VERSION 3.9)
+cmake_minimum_required (VERSION 3.15)
```
Comment (TonyXiang8787) [inline tests/package_tests/CMakeLists.txt:4]: Same here. We don't need 3.15 for this I guess.

File: cmake/export_power_grid_model.cmake (hunk 1 above)
Comment (mgovers) [reply cmake/export_power_grid_model.cmake:5]: oh you're right. good call

**Review by TonyXiang8787 — APPROVED** [review]

**Pull request merged**